### PR TITLE
[BE] CST: 66043 Log claim types to data dog

### PR DESCRIPTION
**Pull request by samcoforma**
## Summary

- added claim type logging to `app/controllers/v0/benefits_claims_controller.rb`
- added rspec test to `spec/controllers/v0/benefits_claims_controller_spec.rb`

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#66043

## Testing done

- Manually verified that it logs claim type data
- added RSpec test and made sure that it passed

## What areas of the site does it impact?
Adds additional logging to the Claims Status Tool

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by jerekshoe — APPROVED** [review]

File: app/controllers/v0/benefits_claims_controller.rb
Comment (jerekshoe) [inline app/controllers/v0/benefits_claims_controller.rb:25]: I'd change this to `claim_type_code` so it matches the others

**Comment by samcoforma** [comment]
@jerekshoe this needs your approval again because it dismissed your review after making the change you suggested. Thanks!

**Review by jerekshoe — APPROVED** [review]

**Pull request merged**
